src/inference/include/openvino/runtime/config.hpp

```diff
@@ -0,0 +1,616 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header for advanced hardware related properties for OpenVINO runtime plugins
+ *        To use in set_config, compile_model, import_model methods
+ *
+ * @file openvino/runtime/config.hpp
+ */
+#pragma once
+
+#include <istream>
+#include <map>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "ie_precision.hpp"
+#include "openvino/core/any.hpp"
+#include "openvino/runtime/common.hpp"
+
+namespace ov {
+
+/**
+ * @brief Enum to define configuration value mutability
+ */
+enum class ConfigMutability {
+    RO,  //!< Read only configuration values can not be passed as input parameter
+    RW,  //!< Read/Write configuration key may change readability in runtime
+    WO,  //!< Write only configuration values could not be obtained from openvino objects
+};
+
+/** @cond INTERNAL */
+namespace util {
+template <typename... Args>
+struct AllConfigs;
+
+template <typename T, typename... Args>
+struct AllConfigs<T, Args...> {
+    constexpr static const bool value =
+        std::is_same<T, std::pair<std::string, ov::Any>>::value && AllConfigs<Args...>::value;
+};
+
+template <typename T>
+struct AllConfigs<T> {
+    constexpr static const bool value = std::is_same<T, std::pair<std::string, ov::Any>>::value;
+};
+
+template <typename T, typename... Args>
+using EnableIfAllConfigs = typename std::enable_if<AllConfigs<Args...>::value, T>::type;
+
+template <typename T, ConfigMutability mutability>
+using EnableIfRaedableConfig =
+    typename std::enable_if<mutability == ConfigMutability::RO || mutability == ConfigMutability::RW, T>::type;
+
+}  // namespace util
+/** @endcond */
+
+/**
+ * @brief This class us used to bind configuration key string with paramter type
+ * @tparam T type of parameter used to pass or get configuration
+ */
+template <typename T, ConfigMutability mutability_ = ConfigMutability::RW>
+struct Key {
+    using value_type = T;                                           //!< Configuration parameter type
+    constexpr static const auto mutability = ConfigMutability::RO;  //!< Configuration parameter readability
```

Comment:
Why it is not initialized from the template parameter `mutability_ `?

Reply:
will be fixed.